### PR TITLE
Integrate gutenberg-mobile release 1.55.2

### PR DESCRIPTION
**Pull request by dcalhoun**
## Description
This PR incorporates the 1.55.2 release of gutenberg-mobile.
For more information about this release and testing instructions, please see the related Gutenberg-Mobile PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/3650

Release Submission Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: RELEASE-NOTES.txt [hunk 1]
```diff
@@ -16,6 +16,7 @@
 * [*] Fix notice overlapping the ActionSheet that displays the More Actions in the Editor. [#16658]
 * [*] The quick action buttons will be hidden when iOS is using a accessibility font sizes. [#16701]
 * [*] Block Editor: Improve unsupported block message for reusable block. [https://github.com/wordpress-mobile/gutenberg-mobile/pull/3621]
+* [**] Fix incorrect block insertion point after blurring the post title field. [https://github.com/wordpress-mobile/gutenberg-mobile/pull/3640]
```
Comment (fluiddot) [inline RELEASE-NOTES.txt]: I see the rest of the entry items related to Gutenberg have the prefix `Block Editor:`, I'm not sure if it's a convention but we might want to include it, wdyt?

**Review by fluiddot — APPROVED** [review]
LGTM 🎊 !

_I only added a comment regarding the release notes but it's just a minor issue._

File: RELEASE-NOTES.txt (hunk 1 above)
Comment (dcalhoun) [reply RELEASE-NOTES.txt]: Addressed in https://github.com/wordpress-mobile/WordPress-iOS/pull/16723/commits/e31f42df3ed55b3698c4397800bc57e914041e75.

**Review by fluiddot — APPROVED** [review]
LGTM 🎊 !

**Comment by mokagio** [comment]
This PR has been shipped as part of 17.6 beta 4 (17.6.0.3) thanks @dcalhoun 🙌

**Pull request merged**